tests/drivers/rtc/rtc_api/testcase.yaml

```diff
@@ -1,11 +1,31 @@
 # Copyright (c) 2022 Bjarki Arge Andreasen
 # SPDX-License-Identifier: Apache-2.0
 
+common:
+  tags:
+    - drivers
+    - rtc
+    - api
+  filter: dt_alias_exists("rtc")
+  depends_on: rtc
+  integration_platforms:
+    - native_posix
 tests:
-  drivers.rtc.rtc_api:
-    tags:
-      - drivers
-      - rtc
-      - api
-    filter: dt_alias_exists("rtc")
-    depends_on: rtc
+  drivers.rtc.rtc_api.basic: {}
+  drivers.rtc.rtc_api.alarm:
+    build_only: true
+    extra_configs:
+      - CONFIG_RTC_ALARM=y
+  drivers.rtc.rtc_api.update:
+    build_only: true
+    extra_configs:
+      - CONFIG_RTC_UPDATE=y
+  drivers.rtc.rtc_api.calibration:
+    build_only: true
+    extra_configs:
+      - CONFIG_RTC_CALIBRATION=y
+  drivers.rtc.rtc_api.full:
+    extra_configs:
+      - CONFIG_RTC_ALARM=y
+      - CONFIG_RTC_UPDATE=y
+      - CONFIG_RTC_CALIBRATION=y
```

Comment:
Actually, it would be easier if all of the `=y` options could just be added to `prj.conf`. I realize that there may be unique code paths where one option is enabled and another is disabled, but this creates a fair bit of code churn for CI, and so far, all drivers support ALARM and CALIBRATE.

Reply:
Fixed.

Reply:
Reverted due to feedback by @bjarki-trackunit - see below.